File: test_regress/t/t_param_in_func_noinline.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2019 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

scenarios(simulator => 1);

top_filename("t/t_param_in_func.v");

compile(
    verilator_flags2 => ["--stats", "+define+NO_INLINE=1"],
    );

execute(check_finished => 1);

# The parameter array should have been put in the constant pool
if ($Self->{vlt_all}) {
    file_grep($Self->{stats}, qr/ConstPool, Tables emitted\s+(\d+)/i, 3);
}

# Shouldn't have any references to the parameter array
foreach my $file (
      glob_all("$Self->{obj_dir}/$Self->{VM_PREFIX}*.h"),
      glob_all("$Self->{obj_dir}/$Self->{VM_PREFIX}*.cpp")
    ) {
    file_grep_not($file, qr/digits/i);
}

ok(1);
1;

File: test_regress/t/t_param_in_func_noinline.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('simulator')
test.top_filename = "t/t_param_in_func.v"

test.compile(verilator_flags2=["--stats", "+define+NO_INLINE=1"])

test.execute()

# The parameter array should have been put in the constant pool
if test.vlt_all:
    test.file_grep(test.stats, r'ConstPool, Tables emitted\s+(\d+)', 3)

# Shouldn't have any references to the parameter array
for filename in (test.glob_some(test.obj_dir + "/" + test.vm_prefix + "*.h") +
                 test.glob_some(test.obj_dir + "/" + test.vm_prefix + "*.cpp")):
    test.file_grep_not(filename, r'digits')

test.passes()
